contracts/MemeFactory.sol

```diff
@@ -419,26 +412,32 @@ abstract contract MemeFactory {
     }
 
     /// @dev Converts collected reference token to OLAS.
-    function scheduleOLASForAscendance(uint256 slippage) external virtual {
+    function scheduleOLASForAscendance(uint256 amount, uint256 slippage) external virtual {
         require(_locked == 1, "Reentrancy guard");
         _locked = 2;
 
         // Slippage limit requirement
         require(slippage <= maxSlippage);
 
-        uint256 localAscendance = scheduledForAscendance;
-        require(localAscendance > 0, "Nothing to burn");
+        if (amount > scheduledForAscendance) {
+            amount = scheduledForAscendance;
+        }
+        require(amount > 0, "Nothing to burn");
 
+        // shouldn't this be inside _buyOLAS as its chain specific?
         // Apply slippage protection
         require(IOracle(oracle).validatePrice(slippage));
 
         // Record msg.sender activity
         mapAccountActivities[msg.sender]++;
 
-        uint256 OLASAmount = _buyOLAS(localAscendance);
+        // Wrap native token to its ERC-20 version, where applicable
```

Comment:
See above, we should wrap the exact amount on demand, not lazy wrap as the state machine for this is complicated.

Reply:
Fine but once again I don't see a complication. All the native needs to be wrapped at the end.